lib/jobs/generate-sku.js

```diff
@@ -118,13 +118,19 @@ function generateSkuJobFactory(
                     });
                 }
 
-                return waterline.nodes.updateByIdentifier(self.nodeId, {
-                    sku: match && match.sku ? match.sku.id : null
-                })
-                .then(function() {
-                    if (match && match.sku) {
-                        return eventsProtocol.publishSkuAssigned(self.nodeId, match.sku.id);
-                    }
+                return waterline.nodes.needByIdentifier(self.nodeId)
+                .then(function(oldNode) {
+                    return waterline.nodes.updateByIdentifier(self.nodeId, {
+                        sku: match && match.sku ? match.sku.id : null
+                    })
+                    .then(function(newNode) {
+                        /* asynchronous */
+                        eventsProtocol.publishNodeAttrEvent(oldNode, newNode, 'sku');
+
+                        if (match && match.sku) {
+                            return eventsProtocol.publishSkuAssigned(self.nodeId, match.sku.id);
```

Comment:
it's previous code,  not intend to change it, and maybe refined to remove this function in near futre